### PR TITLE
fix: add a close button to comment input

### PR DESCRIPTION
**Pull request by tu2463**
## Changes

This PR resolves #1606 https://github.com/dailydotdev/daily/issues/1606

- Added a CloseButton component to the MarkdownInput component, so that we can see a "X" on the top-right corner of the comment box.
image

- Let CommentInputOrModal passes onClose to CommentMarkdownInput as a prop "onRequestClose". This function is eventually passed to MarkdownInput, so that clicking the close button can cancel the comment.

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR
1. Go to a post
2. Click on comment button
3. Should see the comment box with a close button. Clicking the close button will cancel the comment.

-->

### CODE REVIEW

**Comment by jullia02** [comment]
> After this, there should be two components we'll need to update to make this work on desktop. The `MainComment` and `SubComment` components contain a conditional render on the input field we display, we should pass the `onRequestClose` value to be a function that clears the state that manages the input itself.

Thank you for your feedback! We've updated both components to pass the `onRequestClose` prop, so it clears the input state as requested. We've also reviewed your inline comments and incorporated the changes. 

Please let us know if there's anything else you'd like adjusted. 😄

File: packages/shared/src/components/comments/CommentInputOrModal.tsx [hunk 1]
```diff
@@ -11,8 +11,9 @@ import { useMutateComment } from '../../hooks/post/useMutateComment';
 
 interface CommentInputOrModalProps
   extends Partial<LazyModalCommonProps>,
-    Omit<CommentMarkdownInputProps, 'className'> {
+    Omit<CommentMarkdownInputProps, 'className' | 'onRequestClose'> {
```
Comment (sshanzel) [inline packages/shared/src/components/comments/CommentInputOrModal.tsx:14]: Oh, it seems that we won't need the new property after all. Looks like we already have an `onClose` property which we can use.
```suggestion
    Omit<CommentMarkdownInputProps, 'className'> {
```

File: packages/shared/src/components/comments/CommentInputOrModal.tsx [hunk 2]
```diff
@@ -11,8 +11,9 @@ import { useMutateComment } from '../../hooks/post/useMutateComment';
...
   onClose?: () => void;
+  onRequestClose?: () => void;
```
Comment (sshanzel) [inline packages/shared/src/components/comments/CommentInputOrModal.tsx:16]: As mentioned above 🙏 
```suggestion
```

File: packages/shared/src/components/comments/CommentInputOrModal.tsx [hunk 3]
```diff
@@ -23,6 +24,7 @@ interface CommentInputOrModalProps
 export default function CommentInputOrModal({
   onClose,
   className,
+  onRequestClose,
```
Comment (sshanzel) [inline packages/shared/src/components/comments/CommentInputOrModal.tsx:27]:
```suggestion
```

File: packages/shared/src/components/comments/CommentInputOrModal.tsx [hunk 4]
```diff
@@ -42,7 +44,7 @@ export default function CommentInputOrModal({
     <WriteCommentContext.Provider
       value={{ mutateComment: mutateCommentResult }}
     >
-      <CommentMarkdownInput {...props} className={className.input} />
+      <CommentMarkdownInput {...props} className={className.input} onRequestClose={onRequestClose}/>
```
Comment (sshanzel) [inline packages/shared/src/components/comments/CommentInputOrModal.tsx:47]: Now, for this one. We can just use the `onClose` 🥳 
```suggestion
      <CommentMarkdownInput {...props} className={className.input} onRequestClose={onClose}/>
```

File: packages/shared/src/components/comments/MainComment.tsx [hunk 5]
```diff
@@ -175,6 +175,7 @@ export default function MainComment({
             onCommented(...params);
           }}
           onClose={() => onEdit(null)}
+          onRequestClose={() => onEdit(null)}
```
Comment (sshanzel) [inline packages/shared/src/components/comments/MainComment.tsx:178]: Apologies, we can now reuse the `onClose`.
```suggestion
```

File: packages/shared/src/components/comments/MainComment.tsx [hunk 6]
```diff
@@ -187,6 +188,7 @@ export default function MainComment({
             onCommented(...params);
           }}
           onClose={() => onReplyTo(null)}
+          onRequestClose={() => onReplyTo(null)}
```
Comment (sshanzel) [inline packages/shared/src/components/comments/MainComment.tsx:191]:
```suggestion
```

File: packages/shared/src/components/comments/SubComment.tsx [hunk 7]
```diff
@@ -68,6 +68,7 @@ function SubComment({
             onCommented(data, isNew);
           }}
           onClose={() => onEdit(null)}
+          onRequestClose={() => onEdit(null)}
```
Comment (sshanzel) [inline packages/shared/src/components/comments/SubComment.tsx:71]:
```suggestion
```

File: packages/shared/src/components/comments/SubComment.tsx [hunk 8]
```diff
@@ -81,6 +82,7 @@ function SubComment({
             onCommented(...params);
           }}
           onClose={() => onReplyTo(null)}
+          onRequestClose={() => onReplyTo(null)}
```
Comment (sshanzel) [inline packages/shared/src/components/comments/SubComment.tsx:85]:
```suggestion
```

**Review by sshanzel — APPROVED** [review]
Amazing work by both you 👏 👏 👏 

Just the suggested commits, then we should be good to go! 🚀 :shipit:

File: packages/shared/src/components/fields/MarkdownInput/index.tsx [hunk 9]
```diff
@@ -65,6 +66,7 @@ interface MarkdownInputProps
   isLoading?: boolean;
   disabledSubmit?: boolean;
   maxInputLength?: number;
+  onRequestClose?: () => void;
```
Comment (sshanzel) [inline packages/shared/src/components/fields/MarkdownInput/index.tsx:69]:
```suggestion
  onClose?: () => void;
```

File: packages/shared/src/components/fields/MarkdownInput/index.tsx [hunk 10]
```diff
@@ -97,6 +99,7 @@ function MarkdownInput(
     isLoading,
     disabledSubmit,
     maxInputLength,
+    onRequestClose,
```
Comment (sshanzel) [inline packages/shared/src/components/fields/MarkdownInput/index.tsx:102]:
```suggestion
    onClose,
```

File: packages/shared/src/components/fields/MarkdownInput/index.tsx [hunk 11]
```diff
@@ -181,6 +184,13 @@ function MarkdownInput(
           isUptoDate={initialContent === input}
         />
       )}
+      {onRequestClose && (
+        <CloseButton
+          size={ButtonSize.XSmall}
+          className="absolute right-1 top-1 laptop:right-3 laptop:top-3"
+          onClick={onRequestClose}
```
Comment (sshanzel) [inline packages/shared/src/components/fields/MarkdownInput/index.tsx:191]:
```suggestion
      {onClose && (
        <CloseButton
          size={ButtonSize.XSmall}
          className="absolute right-1 top-1 laptop:right-3 laptop:top-3"
          onClick={onClose}
```

File: packages/shared/src/components/comments/CommentInputOrModal.tsx [hunk 12]
```diff
@@ -42,7 +42,11 @@ export default function CommentInputOrModal({
     <WriteCommentContext.Provider
       value={{ mutateComment: mutateCommentResult }}
     >
-      <CommentMarkdownInput {...props} className={className.input} />
+      <CommentMarkdownInput
+        {...props}
+        className={className.input}
+        onRequestClose={onClose}
```
Comment (sshanzel) [inline packages/shared/src/components/comments/CommentInputOrModal.tsx:48]:
```suggestion
        onClose={onClose}
```

**Review by sshanzel — APPROVED** [review]
Just to avoid any conflict. I shall test this thoroughly tomorrow and see if we can merge 🥳

File: packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx [hunk 13]
```diff
@@ -40,6 +40,7 @@ export interface CommentMarkdownInputProps {
   showUserAvatar?: boolean;
   onChange?: (value: string) => void;
   formProps?: FormHTMLAttributes<HTMLFormElement>;
+  onRequestClose?: () => void;
```
Comment (sshanzel) [inline packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx:43]:
```suggestion
  onClose?: () => void;
```

File: packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx [hunk 14]
```diff
@@ -54,6 +55,7 @@ export function CommentMarkdownInputComponent(
     showSubmit = true,
     showUserAvatar = true,
     formProps = {},
+    onRequestClose,
```
Comment (sshanzel) [inline packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx:58]:
```suggestion
    onClose,
```

File: packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx [hunk 15]
```diff
@@ -131,6 +133,7 @@ export function CommentMarkdownInputComponent(
           ) : null
         }
         onValueUpdate={onChange}
+        onRequestClose={onRequestClose}
```
Comment (sshanzel) [inline packages/shared/src/components/fields/MarkdownInput/CommentMarkdownInput.tsx:136]:
```suggestion
        onClose={onClose}
```

File: packages/shared/src/components/fields/MarkdownInput/index.tsx [hunk 16]
```diff
@@ -181,6 +184,13 @@ function MarkdownInput(
           isUptoDate={initialContent === input}
         />
       )}
+      {onClose && (
+        <CloseButton
+          size={ButtonSize.XSmall}
+          className="absolute right-1 top-1 laptop:right-3 laptop:top-3"
```
Comment (sshanzel) [inline packages/shared/src/components/fields/MarkdownInput/index.tsx:190]:
```suggestion
          className="absolute right-3 top-3"
```

File: packages/shared/src/components/fields/MarkdownInput/index.tsx [hunk 17]
```diff
@@ -181,6 +184,13 @@ function MarkdownInput(
           isUptoDate={initialContent === input}
         />
       )}
+      {onClose && (
+        <CloseButton
+          size={ButtonSize.XSmall}
+          className="absolute right-3 top-3"
```
Comment (sshanzel) [inline packages/shared/src/components/fields/MarkdownInput/index.tsx:190]:
```suggestion
          size={ButtonSize.Small}
          className="absolute right-3 top-2"
```

**Comment by sshanzel** [comment]
@tu2463 @jullia02 thank you both for your efforts. We've now merged your work, and should be seen within the app in a few minutes 🥳 🙏

**Pull request merged**